bgpd/bgp_conditional_adv.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * BGP Conditional advertisement
+ * Copyright (C) 2020  Samsung Research Institute Bangalore.
+ *			Madhurilatha Kuruganti
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "bgpd/bgp_conditional_adv.h"
+
+const char *get_afi_safi_str(afi_t afi, safi_t safi, bool for_json);
+
+static route_map_result_t
+bgp_check_rmap_prefixes_in_bgp_table(struct bgp_table *table,
+				     struct route_map *rmap)
+{
+	struct attr dummy_attr = {0};
+	struct bgp_dest *dest;
+	struct bgp_path_info *pi;
+	struct bgp_path_info path = {0};
+	struct bgp_path_info_extra path_extra = {0};
+	const struct prefix *dest_p;
+	route_map_result_t ret = RMAP_DENYMATCH;
+
+	for (dest = bgp_table_top(table); dest; dest = bgp_route_next(dest)) {
+		dest_p = bgp_dest_get_prefix(dest);
+		assert(dest_p);
+
+		for (pi = bgp_dest_get_bgp_path_info(dest); pi; pi = pi->next) {
+			dummy_attr = *pi->attr;
+
+			/* Fill temp path_info */
+			prep_for_rmap_apply(&path, &path_extra, dest, pi,
+					    pi->peer, &dummy_attr);
+
+			RESET_FLAG(dummy_attr.rmap_change_flags);
+
+			ret = route_map_apply(rmap, dest_p, RMAP_BGP, &path);
+			if (ret != RMAP_PERMITMATCH)
+				bgp_attr_flush(&dummy_attr);
+			else {
+				zlog_debug(
+					"%s: Condition map routes present in BGP table",
+					__func__);
+				return ret;
+			}
+		}
+	}
+
+	zlog_debug("%s: Condition map routes not present in BGP table",
```

Comment:
here as well, we need log if explicit logging is enabled

Reply:
do you mean to add [no] debug bgp conditional-advertisement ?

Reply:
no, I mean at least if `debug bgp updates` enabled.

Reply:
ok i will add the check if (BGP_DEBUG(update, UPDATE_OUT))

Reply:
addressed this comment